api/http/handler/endpoints/endpoint_create.go

```diff
@@ -40,6 +40,7 @@ type endpointCreatePayload struct {
 	AzureAuthenticationKey string
 	TagIDs                 []portainer.TagID
 	EdgeCheckinInterval    int
+	ContainerEngine        string
```

Comment:
Please add this parameter to the API metadata in the comments section.

Also update the validation section to only accept a value that is one of the defined constants (see the comment for the `portainer.go` file about that).

Reply:
Updated